src/test/c_api_stress_tests.cpp

```diff
@@ -0,0 +1,441 @@
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiGetMetadataInALoop,
+        &StressCapiConfigChanges::changeToAutoShape,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, RemovePipelineDefinitionDuringGetMetadataLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK,
+        StatusCode::MODEL_NAME_MISSING};  // when pipeline is retired
+    std::set<StatusCode> allowedLoadResults = {};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiGetMetadataInALoop,
+        &StressCapiConfigChanges::removePipelineDefinition,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, ChangedPipelineConnectionNameDuringGetMetadataLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {StatusCode::MODEL_NAME_MISSING};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiGetMetadataInALoop,
+        &StressCapiConfigChanges::changeConnectionName,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, AddedNewPipelineDuringGetMetadataLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiGetMetadataInALoop,
+        &StressCapiConfigChanges::addNewPipeline,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, RetireSpecificVersionUsedDuringGetMetadataLoad) {
+    // we declare specific version used (1) and latest model version policy with count=1
+    // then we add version 2 causing previous default to be retired
+    SetUpConfig(stressTestPipelineOneDummyConfigSpecificVersionUsed);
+    bool performWholeConfigReload = false;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK,  // we expect full continuity of operation
+        StatusCode::MODEL_NAME_MISSING};                         // we hit when all config changes finish to propagate
+    std::set<StatusCode> allowedLoadResults = {};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiGetMetadataInALoop,
+        &StressCapiConfigChanges::retireSpecificVersionUsed,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+
+TEST_F(StressCapiConfigChanges, AddModelDuringGetModelStatusLoad) {
+    bool performWholeConfigReload = true;  // we just need to have all model versions rechecked
```

Comment:
Why we have different ret codes than in 
StressModelConfigChanges, AddModelDuringGetModelStatusLoad?